packages/holocron/src/promiseStore.jsx

```diff
@@ -0,0 +1,87 @@
+import { useDispatch } from 'react-redux';
+
+/** @typedef {import('redux-thunk').ThunkAction} ThunkAction */
+
+/**
+ * A class to store promises by domain and key
+ *
+ * In general, Module code should not interact with the promise store directly,
+ *   instead dispatching the below thunks, or using the provided hooks
+ */
+class PromiseStore {
+  constructor() {
+    this.promises = {};
```

Comment:
Can we make is private?

Reply:
and maybe a `Map`? if not a [null-prototype object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object#null-prototype_objects:~:text=In%20practice%2C%20objects%20with%20null%20prototype%20are%20usually%20used%20as%20a%20cheap%20substitute%20for%20maps.) (`Object.create(null)`)

Reply:
Addressed both in [cbe1f89](https://github.com/americanexpress/holocron/pull/178/commits/cbe1f89748c2c1579105cf1aede9a809b27ac2a3)